nodes/src/nodes/guardrails/IInstance.py

```
@@ -0,0 +1,156 @@
+            question: The incoming Question object.
+        """
+        engine = self.IGlobal.engine
+
+        # Deep copy to prevent mutation of the original
+        question = copy.deepcopy(question)
+
+        # Collect question text for evaluation
+        text_parts = []
+        if question.questions:
+            for q in question.questions:
+                text_parts.append(q.text)
```

Ruff (0.15.7)

[warning] Use `list.extend` to create a transformed list

(PERF401)